test/ServiceManagerTest.php

```diff
@@ -364,4 +365,51 @@ public function testResolvedAliasNoMatchingAbstractFactoryReturnsFalse()
 
         self::assertFalse($serviceManager->has('Alias'));
     }
+
+    /**
+     * Hotfix #3
+     * @see https://github.com/laminas/laminas-servicemanager/issues/3
+     */
+    public function testConfigureMultipleTimesAvoidsDuplicates()
+    {
+        $delegatorFactory = function (
+            ContainerInterface $container,
+            $name,
+            callable $callback
+        ) {
+            /** @var InvokableObject $instance */
+            $instance = $callback();
+            $options = $instance->getOptions();
+            $inc = $options['inc'] ?? 0;
+            return new InvokableObject(['inc' => ++$inc]);
+        };
+
+        $config = [
+            'factories' => [
+                'Foo' => function () {
+                    return new InvokableObject();
+                },
+            ],
+            'delegators' => [
+                'Foo' => [
+                    $delegatorFactory,
+                    LazyServiceFactory::class,
+                ],
+            ],
+            'lazy_services' => [
```

Comment:
This test is verifying both the `lazy_services` and the `delegator` changes: should probably be split up into two tests

Reply:
I think this is OK; it's in the scope of the bugfix imo.